src/runtime_src/core/common/info_aie.cpp

```diff
@@ -315,12 +320,87 @@ populate_aie_shim(const xrt_core::device *device, const std::string& desc)
       if (oshim.find("event") != oshim.not_found())
         addnodelist("event", "events", oshim, ishim);
 
-      tile_array.push_back(std::make_pair("tile" + std::to_string(col), ishim));
+      tile_array.push_back({"tile" + std::to_string(col), ishim});
     }
 
     pt.add_child("tiles", tile_array);
 
-  } catch (const std::exception& ex){
+  }
+  catch (const std::exception& ex) {
+    pt.put("error_msg", (boost::format("%s %s") % ex.what() % "found in the AIE shim"));
+  }
+
+  return pt;
+}
+
+// Populate the AIE Mem tile information from the input XRT device
+boost::property_tree::ptree
+populate_aie_mem(const xrt_core::device* device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  pt.put("description", desc);
+  boost::property_tree::ptree pt_mem;
+
+  try {
```

Comment:
A try catch block in a try catch block looks very suspicious to me. Especially with the complicated catch for the no_such_query case

We should not rely on try catch blocks to conduct normal operations. Just to catch exceptions and handle the output.

The better way to handle this would be to place the try catch blocks sequentially and try both. Then to validate the output.

```

// Read AIE Mem information from edge device
try {
  std::string aie_data = xrt_core::device_query<qr::aie_mem_info_sysfs>(device);
  std::stringstream ss(aie_data);
  boost::property_tree::read_json(ss, pt_mem);
}
catch (const xrt_core::query::no_such_key&) {
  // Not edge device
}


// Read AIE Mem information of pcie device
try {
  asd_parser::aie_tiles_info tiles_info{0};
  // On Pcie platforms use driver calls to get AIE memm info
  pt_mem = asd_parser::get_formated_tiles_info(device, asd_parser::aie_tile_type::mem, tiles_info);
}
catch (const xrt_core::query::no_such_key&) {
  // Not pcie device
}

// Check that the aie data was populated
// Otherwise we are on an unsupported device for aie
if (pt_mem.empty()) {
  pt.put("error_msg", ex.what());
  return pt;
}
```

Reply:
Thanks @dbenusov-xilinx, made changes to reflect this.